### PR TITLE
Add Python Turtle workshop

### PR DESCRIPTION
**Pull request by YashKalbande**
Python turtle workshop introduce turtle library of python to run simple yet powerful code to create exciting visual .

### CODE REVIEW

File: workshops/python turtal/README.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+author: '@YashKalbande'
+---
+
+
+# Exciting Visuals with Python Turtle
+
+
+## Introduction
+
+Turtle graphics is a Python feature like a drawing board, which lets us command a turtle to draw all over it. It is a popular way for introducing programming to kids. Virtual turtles can be programmed to move around the screen. The turtle draws lines as it moves. The "turtle" could look like the turtle animal, an arrow, or be invisibile. The user can write turtle programs that draw beautiful shapes and learn to program at the same time.The original Turtle software was developed by Wally Feurzig and Seymour Papert in 1966.<br/>
```
Comment (bezlin6mechminerz) [inline workshops/python turtal/README.md:13]:
```suggestion
Turtle graphics is a Python feature like a drawing board, which lets us command a turtle to draw all over it. It is a popular way of introducing programming to kids. Virtual turtles can be programmed to move around the screen. The turtle draws lines as it moves. The "turtle" could look like the turtle animal, an arrow, or be invisible. The user can write turtle programs that draw beautiful shapes and learn to program at the same time. The original Turtle software was developed by Wally Feurzig and Seymour Papert in 1966.<br/>
```

File: workshops/python turtal/README.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+2. Turn 90 degrees to the left.
+3. Move forward 100 steps.
+4. Turn 90 degrees to the left.
+5. Move forward 100 steps.
+6. Turn 90 degrees to the left.
+7. Move forward 100 steps. The turtle has ended up where it started.
+
+With these seven steps, the turtle draws a square. The `from turtle import *` is an instruction needed at the beginning of all of your turtle programs. It imports the turtle module so you can do the turtle instructions.
+
+There are many instructions like left() and forward(). These instructions are called functions. This tutorial explains many of the functions in the turtle module. When you learn more of these functions, you will be able to draw many different shapes and beautiful pictures!
```
Comment (bezlin6mechminerz) [inline workshops/python turtal/README.md:69]:
```suggestion
There are many instructions like left() and forward(). These instructions are called functions. This tutorial explains many of the functions in the turtle module. When you learn more about these functions, you will be able to draw many different shapes and beautiful pictures!
```

File: workshops/python turtal/README.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+## Turtle Motion
+
+By calling these functions, the turtle can be made to move around the screen. Imagine the turtle holding a pen down on the ground and drawing a line as it moves around.
+
+The turtle's position is two numbers: the X coordinate and Y coordinate.
+
+### forward(*distance*)
+
+The forward() function moves the turtle *distance* number of steps in the current direction. If the pen is down (see pendown() and penup()) a line will be drawn as the turtle moves forward. If *distance* is a negative number, the turtle will move backwards.
```
Comment (bezlin6mechminerz) [inline workshops/python turtal/README.md:80]:
```suggestion
The forward() function moves the turtle *distance* a number of steps in the current direction. If the pen is down (see pendown() and penup()) a line will be drawn as the turtle moves forward. If *distance* is a negative number, the turtle will move backward.
```

**Review by sampoder — CHANGES_REQUESTED** [review]
Could we make this workshop flow more and not just be a guide? Let's have a main goal and not so may different little sections

**Comment by Hugoyhu** [comment]
Hi, could I please have your Slack handle and/or a Hack Club you lead? Thanks!

**Comment by YashKalbande** [comment]
@Hugoyhu my hack club handle is "yash.kalbande"

**Comment by Hugoyhu** [comment]
Please advise on which hack club you are a leader of? Thanks

On Oct 7, 2020, at 10:46 PM, Yash.Kalbnde <notifications@github.com<mailto:notifications@github.com>> wrote:



@Hugoyhu<https://github.com/Hugoyhu> my hack club handle is "yash.kalbande"

File: workshops/python turtal/README.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+for i in range(num_sides): 
+    polygon.forward(side_length) 
+    polygon.right(angle) 
+      
+turtle.done()  
+```
+![](https://cloud-36kk4uh0h.vercel.app/hexagon.gif)
+
+## Rainbow Benzene 
+So now you have gain basic knowledge of python turtle library. Now lets build the exicting raindow benzene with turtle.
```
Comment (lachlanjc) [inline workshops/python turtal/README.md:232]: Grammar/punctuation is off here

**Review by lachlanjc — CHANGES_REQUESTED** [review]
This is an awesome start, thank you! Generally, I think this workshop reads more like documentation with illustrations for the Turtle library than a friend showing you how to build something amazing, the latter of which I think is a critical part of a successful workshop. I think this workshop could use a bit of revision to make it clearer what’s being made at the beginning—I didn’t really understand what a turtle is or what I can do with one—and be clearer about next steps at tthe end. There’s also a handful of typos & grammatical mistakes throughout it, & missing alt text descriptions on the images/GIFs throughout. The path of the workshop also has typos—it should be `python_turtle`.

I’d like to move forward with this workshop, but these issues will need to be addressed. Thanks!

**Comment by faisalsayed10** [comment]
Hi, I see that this doesn't fulfil all the requirements of the workshop bounty guidelines as the demo meeting with 3+ members is missing. Please take a look into it

File: workshops/python_turtle/README.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+description: 'Exciting Visuals with Python Turtle'
+author: '@YashKalbande'
+---
+
+
+# Exciting Visuals with Python Turtle
+
+
+## Introduction
+Welcome! my friends.  My name is Yash, and I am a technology enthusiast. Have you ever heard that you can create exciting visuals with help of Python?  My goal is to help you understand the python turtle library. With the help of the turtle library, we can create stunning visuals. First, I'm going to present to you the basics of the python turtle library. Then I'll share with you shapes like Triangle, Star, Hexagon. Finally, I'll ask you to create some more visual using ideas given in the end.
```
Comment (lachlanjc) [inline workshops/python_turtle/README.md:12]: Don’t introduce yourself—this should stand alone as a tutorial

File: workshops/python_turtle/README.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+
+![Benzen ring with Turtle](https://cloud-36kk4uh0h.vercel.app/turtle_benzene_ring.gif)
+
+
+## Quickstart
+
+TODO - Explain how to install and setup Python 3 on [repl.it](https://repl.it/languages/python3) for turtle graphics.
```
Comment (lachlanjc) [inline workshops/python_turtle/README.md:19]: There shouldn’t be TODO

File: workshops/python_turtle/README.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+### end_fill()
+
+The end_fill() function will stop recording the moves for the filled-in shape and draw the shape.
+
+To get more information about color [visit hear](https://docs.python.org/3/library/turtle.html#turtle.color)
+
+## Hacking
+- There are endless ways to make this turtle completely your own! Go back through your code and look for anything that can be played with. Try changing the default values of your forward, angle, color, size, etc.
+
+- Here are some examples to give you some more idea:
```
Comment (lachlanjc) [inline workshops/python_turtle/README.md:344]:
```suggestion

There are endless ways to make this turtle completely your own! Go back through your code and look for anything that can be played with. Try changing the default values of your forward, angle, color, size, etc.

Here are some examples to give you some ideas:
```

File: workshops/python_turtle/README.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+
+### begin_fill()
+
+The begin_fill() starts recording the moves that will be the outline of the filled-in shape. The filled-in shape will not be drawn until end_fill() is called.
+
+### end_fill()
+
+The end_fill() function will stop recording the moves for the filled-in shape and draw the shape.
+
+To get more information about color [visit hear](https://docs.python.org/3/library/turtle.html#turtle.color)
```
Comment (lachlanjc) [inline workshops/python_turtle/README.md:339]:
```suggestion
To get more information about color, [visit here](https://docs.python.org/3/library/turtle.html#turtle.color).
```

**Review by lachlanjc — CHANGES_REQUESTED** [review]
This is a solid improvement Yash! Right now this feels more like a visual demo of documentation of the library’s functions than focused around building something real-world/concrete, but to be honest I’m not quite sure what to suggest there, & I do want us to have a solid intro to this library. Left a few quick comments before we merge. We also need the video demo Fayd mentioned.

File: workshops/python_turtle/README.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+7. Move forward 100 steps. The turtle has ended up where it started.
+
+With these seven steps, the turtle draws a square. The `from turtle import *` is an instruction needed at the beginning of all of your turtle programs. It imports the turtle module so you can do the turtle instructions.
+
+There are many instructions like left() and forward(). These instructions are called functions. This tutorial explains many of the functions in the turtle module. When you learn more about these functions, you will be able to draw many different shapes and beautiful pictures!
+
+There are many instructions like left() and forward(). These instructions are called functions. Commonly used turtle methods are :
+
+| Method        | Parameter           | DESCRIPTION  | Example |
+| ------------- |:-------------:| -----:| --------:|
```
Comment (lachlanjc) [inline workshops/python_turtle/README.md:56]:
```suggestion
| Method        | Parameter           | Description  | Example |
| ------------- | ------------- | ----- | -------- |
```

File: workshops/python_turtle/README.md (hunk 6 above)
Comment (sampoder) [reply workshops/python_turtle/README.md:17]: fyi @YashKalbande, you don't need to install python3 it's preinstalled :)

File: workshops/python_turtle/README.md (hunk 5 above)
Comment (YashKalbande) [reply workshops/python_turtle/README.md]: done.

File: workshops/python_turtle/README.md (hunk 6 above)
Comment (YashKalbande) [reply workshops/python_turtle/README.md]: done

**Pull request merged**